frontend/packages/process-editor/src/hooks/useBpmnEditor.ts

```diff
@@ -40,87 +73,39 @@ export const useBpmnEditor = (): UseBpmnViewerResult => {
         await modelerInstance.importXML(bpmnXml);
         const canvas: any = modelerInstance.get('canvas');
         canvas.zoom('fit-viewport');
-        // Reset the dataTasks tracking states when the editor is initialized
-        setDataTasksAdded([]);
-        setDataTasksRemoved([]);
       } catch (exception) {
         console.log('An error occurred while rendering the viewer:', exception);
       }
     };
 
     initializeEditor();
-  }, [
-    bpmnXml,
-    modelerRef,
-    setBpmnDetails,
-    setNumberOfUnsavedChanges,
-    setDataTasksAdded,
-    setDataTasksRemoved,
-  ]);
+  }, []);
```

Comment:
Is it on purpose that this array is without dependencies? 😄 Maybe we can add a comment explaining why?

Reply:
There is something with the bpmn-modeler in combination with the useEffects that goes totally mad when adding the dependencies 😅 
But David made some changes addig a singleton canvas in a PR recently so it might be possible to add them now 🤔 I can check! 
If not, I will add a comment!

Reply:
Aah, I remember having the same issue when I worked there some months ago! 
If the issue is still there, I suggest that we add a comment just so that the next people working in the code will understand why 😄

Reply:
Added a few dependencies where it was possible. And added comment where not all could be added 🫡